src/ngraph/pass/constant_propagation.cpp

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
+* Copyright 2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "constant_propagation.hpp"
```

Comment:
Typically, I don't care much about naming, but, in this particular case, I think *constant propagation* is a rather confusing name since it means a very specific optimization in compilers which is to propagate the *same* constant values down assignment chains. This optimization, on the other hand, *transforms* original constant values. What does @diyessi think?

Reply:
I think I like constant folding